xbmc/utils/SaveFileStateJob.cpp

```diff
@@ -162,7 +171,14 @@ void CSaveFileState::DoWork(CFileItem& item,
           if (!videodatabase.GetStreamDetails(dbItem) ||
               dbItem.GetVideoInfoTag()->m_streamDetails != item.GetVideoInfoTag()->m_streamDetails)
           {
-            videodatabase.SetStreamDetailsForFile(item.GetVideoInfoTag()->m_streamDetails, progressTrackingFile);
+            const int idFile = videodatabase.SetStreamDetailsForFile(
+                item.GetVideoInfoTag()->m_streamDetails, item.GetDynPath());
```

Comment:
@78andyp Why use item.GetDynPath() instead of progressTrackingFile? This bypasses other logic at the top of the function.
For the specific case of bluray you're setting progressTrackingFile to item.GetDynPath()
Got curious while stepping through this for pvr

Reply:
The logic has changed further - see https://github.com/xbmc/xbmc/pull/25140/files
I'll have to go back and have a play to see why I made that specific change.
I think it's something along the lines that `progressTrackingFile` points to the DynPath() before the bluray:// vfs path including the playlist is derived.